python/cudf/cudf/_lib/unaryops.pyx

```diff
@@ -156,25 +150,16 @@ def nans_to_nulls(py_col):
         result = cpp_unaryops.nans_to_nulls(c_col[0])
 
     mask = None
-    if result.first:
+    if result.second:
         mask_ptr = int(<uintptr_t>result.first)
-        mask = rmm.device_array_from_ptr(
-            mask_ptr,
-            nelem=calc_chunk_size(len(py_col), mask_bitsize),
-            dtype=mask_dtype,
-            finalizer=rmm._make_finalizer(mask_ptr, 0)
-        )
+        mask = Buffer(ptr=mask_ptr, size=len(py_col))
```

Comment:
I imagine this API should be updated to return a `device_buffer` instead of a raw pointer, but in the meantime this should be updated to construct a `DeviceBuffer` from the raw pointer and then pass that to the `Buffer(...)`

Reply:
Good catch

Reply:
`Buffer(...)` just does this underneath; is there something I've missed?

Reply:
As far as I know `Buffer(...)` doesn't do this underneath. It will just set the `ptr` attribute to `mask_ptr`, `size` attribute to `len(py_col)` and `owner` attribute to `None`. This memory will never be freed with the current code.